### PR TITLE
Change underscore to lodash

### PR DESCRIPTION
**Pull request by tahmeedshah**

### CODE REVIEW

**Comment by blaisefonseca** [comment]
@tahmeedshah if the PR is ready please change status to Ready for review

**Comment by blaisefonseca** [comment]
was the boot.js updated to use lodah@4.*?
image

File: app/views/report-analyzer/reports.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
                 };
                 sessionStorage.setItem('nrAnalyzerData', JSON.stringify(data));
 
-                $location.url(_.trimRight($location.path(), '/') + '/analyzer');
+                $location.url(_.forInRight($location.path(), '/') + '/analyzer');
```
Comment (blaisefonseca) [inline app/views/report-analyzer/reports.js:24]: wrong function, it should be https://lodash.com/docs/4.17.15#trimEnd

**Comment by blaisefonseca** [comment]
remove underscore factory from service.js
image

**Comment by blaisefonseca** [comment]
https://github.com/lodash/lodash/wiki/Migrating

File: app/js/services.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ define(['app', 'lodash', 'linqjs'], function (app, _, Enumerable) {
 		};
 	});
 
-	app.factory('underscore', [function() {
+	app.factory('lodash', [function() {
 		return _;
 	}])
```
Comment (blaisefonseca) [inline app/js/services.js:15]: no need for the factory please remove

File: app/views/forms/edit/edit-capacityBuildingResource.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 define(['app', 'views/forms/edit/edit', '../view/view-capacity-building-resource.directive'], function (app) {
 
-  app.controller("editCapacityBuildingResource", ["$scope", "$http", "$filter", "$q", "$routeParams", "$controller","$location", "Thesaurus", "Enumerable", "underscore", function ($scope, $http, $filter, $q, $routeParams, $controller,$location, thesaurus, Enumerable, _) {
+  app.controller("editCapacityBuildingResource", ["$scope", "$http", "$filter", "$q", "$routeParams", "$controller","$location", "Thesaurus", "Enumerable", "lodash", function ($scope, $http, $filter, $q, $routeParams, $controller,$location, thesaurus, Enumerable, _) {
```
Comment (blaisefonseca) [inline app/views/forms/edit/edit-capacityBuildingResource.js:4]: remove lodash from controller and put inside define

File: app/views/forms/edit/permit-selection-directive.html.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ app.directive("existingPermit", [ function () {
 		link : function($scope, $element, $attrs) {
 
         },
-		controller : ["$scope", "$http", "Thesaurus", "$filter", "underscore", "guid",
+		controller : ["$scope", "$http", "Thesaurus", "$filter", "lodash", "guid",
```
Comment (blaisefonseca) [inline app/views/forms/edit/permit-selection-directive.html.js:21]: remove from controller and put inside define

File: app/views/forms/edit/permit-selection-directive.js [hunk 5]
```diff
@@ -18,7 +18,7 @@ app.directive("existingPermit", [ function () {
 		link : function($scope, $element, $attrs) {
 
         },
-		controller : ["$scope", "$http", "Thesaurus", "$filter", "underscore", "guid",
+		controller : ["$scope", "$http", "Thesaurus", "$filter", "lodash", "guid",
```
Comment (blaisefonseca) [inline app/views/forms/edit/permit-selection-directive.js:21]: remove from controller and put inside define

File: app/views/forms/edit/warning-message-cna.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ define(['app', 'underscore', "text!views/forms/edit/warning-message-cna.html", '
            scope: {
                 doc:'='
            },
-           controller: ["$scope", "underscore", "searchService", "solr","$q",
+           controller: ["$scope", "lodash", "searchService", "solr","$q",
```
Comment (blaisefonseca) [inline app/views/forms/edit/warning-message-cna.js:16]: remove from controller and put inside define

File: app/views/forms/view/abs/view-abs-checkpoint-communique.directive.js [hunk 7]
```diff
@@ -19,7 +18,7 @@ app.directive("viewAbsCheckpointCommunique", [function () {
 		{
 			$scope.showPdf = $attr.showPdf === undefined || $attr.showPdf != "false";
 		},
-		controller: ["$scope", "IStorage", "$http", "$q","underscore","realm", function ($scope, storage, $http, $q, _, realm)
+		controller: ["$scope", "IStorage", "$http", "$q","lodash","realm", function ($scope, storage, $http, $q, _, realm)
```
Comment (blaisefonseca) [inline app/views/forms/view/abs/view-abs-checkpoint-communique.directive.js:21]: remove from controller and put inside define

File: app/views/forms/view/abs/view-abs-national-report.directive.js [hunk 8]
```diff
@@ -14,7 +14,7 @@ app.directive("viewAbsNationalReport", [function () {
 			target  : "@linkTarget",
 			hide	: "@"
 		},
-		controller : ["$scope", "underscore", "commonjs", "$filter", function ($scope, _, commonjs, $filter)
+		controller : ["$scope", "lodash", "commonjs", "$filter", function ($scope, _, commonjs, $filter)
```
Comment (blaisefonseca) [inline app/views/forms/view/abs/view-abs-national-report.directive.js:17]: remove from controller and put inside define

File: app/views/forms/view/abs/view-abs-permit.directive.js [hunk 9]
```diff
@@ -21,7 +21,7 @@ app.directive("viewAbsPermit", [function () {
 			$scope.contacts = undefined;
 			$scope.showPdf = $attr.showPdf === undefined || $attr.showPdf != "false";
 		},
-		controller: ["$scope", "IStorage", "$http", "$q","underscore", function ($scope, storage, $http, $q, _)
+		controller: ["$scope", "IStorage", "$http", "$q","lodash", function ($scope, storage, $http, $q, _)
```
Comment (blaisefonseca) [inline app/views/forms/view/abs/view-abs-permit.directive.js:24]: remove from controller and put inside define

File: app/views/help/forms/edit-help.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
-define(['app', 'underscore', 'ngMaterial', 'ngAria', 'angular-animate', 'components/scbd-angularjs-services/services/main', 'components/scbd-angularjs-services/filters/scbd-filters', 'components/scbd-angularjs-controls/form-control-directives/all-controls',
+define(['app', 'lodash', 'ngMaterial', 'ngAria', 'angular-animate', 'components/scbd-angularjs-services/services/main', 'components/scbd-angularjs-services/filters/scbd-filters', 'components/scbd-angularjs-controls/form-control-directives/all-controls',
 	'/app/js/common'], function (app, _) {
 		app.controller("editHelpController",
-			["$routeParams", "$scope", "$rootScope", "$q", "underscore", '$http',  '$location', 'commonjs', '$mdDialog',
+			["$routeParams", "$scope", "$rootScope", "$q", "lodash", '$http',  '$location', 'commonjs', '$mdDialog',
```
Comment (blaisefonseca) [inline app/views/help/forms/edit-help.js:4]: remove lodash from controller and put inside define

File: app/views/help/presentations/presentation.js [hunk 11]
```diff
@@ -4,7 +4,7 @@ define(['app'], function (app) {
 //require("app", "dragAndDrop")
 
 app.controller("presentationController",
-	["$rootScope", "$location" , "$scope", "$q", "$window", "IStorage", "underscore",
+	["$rootScope", "$location" , "$scope", "$q", "$window", "IStorage", "lodash",
```
Comment (blaisefonseca) [inline app/views/help/presentations/presentation.js:7]: remove lodash from controller and put inside define

File: app/views/help/releases.js [hunk 12]
```diff
@@ -4,9 +4,9 @@ define(['app'], function (app) {
 //require("app", "dragAndDrop")
 
 app.controller("presentationController",
-	["$rootScope", "$scope", "$q", "underscore",'$http',function ($rootScope, $scope, $q, _, $http) {	
-	
-			
+   ["$rootScope", "$scope", "$q", "lodash",'$http',function ($rootScope, $scope, $q, _, $http) {
```
Comment (blaisefonseca) [inline app/views/help/releases.js:7]: remove lodash from controller and put inside define

**Review by blaisefonseca — CHANGES_REQUESTED** [review]
some more fixes are required

File: app/views/help/releases.js (hunk 12 above)
Comment (tahmeedshah) [reply app/views/help/releases.js:10]: moved lodash from controller to define

File: app/views/report-analyzer/reports.js (hunk 1 above)
Comment (tahmeedshah) [reply app/views/report-analyzer/reports.js]: changed to trimEnd

File: app/js/services.js (hunk 2 above)
Comment (tahmeedshah) [reply app/js/services.js]: factory  removed

File: app/views/forms/edit/edit-capacityBuildingResource.js (hunk 3 above)
Comment (tahmeedshah) [reply app/views/forms/edit/edit-capacityBuildingResource.js]: moved lodash from controller to define

File: app/views/forms/edit/permit-selection-directive.html.js (hunk 4 above)
Comment (tahmeedshah) [reply app/views/forms/edit/permit-selection-directive.html.js]: moved lodash from controller to define

File: app/views/forms/edit/permit-selection-directive.js (hunk 5 above)
Comment (tahmeedshah) [reply app/views/forms/edit/permit-selection-directive.js]: moved lodash from controller to define

File: app/views/forms/edit/warning-message-cna.js (hunk 6 above)
Comment (tahmeedshah) [reply app/views/forms/edit/warning-message-cna.js]: moved lodash from controller to define

File: app/views/forms/view/abs/view-abs-checkpoint-communique.directive.js (hunk 7 above)
Comment (tahmeedshah) [reply app/views/forms/view/abs/view-abs-checkpoint-communique.directive.js]: moved lodash from controller to define

File: app/views/forms/view/abs/view-abs-national-report.directive.js (hunk 8 above)
Comment (tahmeedshah) [reply app/views/forms/view/abs/view-abs-national-report.directive.js]: moved lodash from controller to define

File: app/views/forms/view/abs/view-abs-permit.directive.js (hunk 9 above)
Comment (tahmeedshah) [reply app/views/forms/view/abs/view-abs-permit.directive.js]: moved lodash from controller to define

File: app/views/help/forms/edit-help.js (hunk 10 above)
Comment (tahmeedshah) [reply app/views/help/forms/edit-help.js]: moved lodash from controller to define

File: app/views/help/presentations/presentation.js (hunk 11 above)
Comment (tahmeedshah) [reply app/views/help/presentations/presentation.js]: moved lodash from controller to define

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**